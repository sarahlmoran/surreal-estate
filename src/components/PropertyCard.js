import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import "../styles/property-card.css";

const PropertyCard = (props) => {
  const { bathrooms, bedrooms, city, email, price, title, type } = props;
  return (
    <div className="add-property">
      <FontAwesomeIcon icon="house" />
      <div className="property-card__title">{title}</div>
      <div className="property-card__city">{city}</div>
      <div className="property-card__type">{type}</div>
      <div className="property-card__bedrooms">{bedrooms}</div>
      <FontAwesomeIcon icon="bed" />
      <div className="property-card__bathrooms">{bathrooms}</div>
      <FontAwesomeIcon icon="bath" />
      <FontAwesomeIcon icon="sterling-sign" />
      <div className="property-card__price">{price}</div>
      <div className="property-card__email">
        <a href={`mailto:${email}`}>Email</a>
      </div>
      <FontAwesomeIcon icon="envelope" />
    </div>
  );
};

export default PropertyCard;
