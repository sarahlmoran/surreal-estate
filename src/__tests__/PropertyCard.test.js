import React from "react";
import { render } from "@testing-library/react";
import PropertyCard from "../components/PropertyCard";

describe("PropertyCard", () => {
  const validProps = {
    bathrooms: 2,
    bedrooms: 4,
    city: "Stub City",
    email: "stub@email.com",
    price: "120,000",
    title: "Stub title",
    type: "flat",
  };

  it("renders", () => {
    const { asFragment } = render(
      <PropertyCard
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        city={validProps.city}
        email={validProps.email}
        price={validProps.price}
        title={validProps.title}
        type={validProps.type}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(
      <PropertyCard
        bathrooms={validProps.bathrooms}
        bedrooms={validProps.bedrooms}
        city={validProps.city}
        email={validProps.email}
        price={validProps.price}
        title={validProps.title}
        type={validProps.type}
      />
    );
    expect(getByText("2")).toHaveClass("property-card__bathrooms");
    expect(getByText("4")).toHaveClass("property-card__bedrooms");
    expect(getByText("Stub City")).toHaveClass("property-card__city");
    expect(getByText("stub@email.com")).toHaveClass("property-card__email");
    expect(getByText("120,000")).toHaveClass("property-card__price");
    expect(getByText("Stub title")).toHaveClass("property-card__title");
    expect(getByText("flat")).toHaveClass("property-card__type");
  });
});
