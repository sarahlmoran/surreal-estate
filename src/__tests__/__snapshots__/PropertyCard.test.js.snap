// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PropertyCard renders 1`] = `
<DocumentFragment>
  <div
    class="add-property"
  >
    <div
      class="property-card__bathrooms"
    >
      2
    </div>
    <div
      class="property-card__bedrooms"
    >
      4
    </div>
    <div
      class="property-card__city"
    >
      Stub City
    </div>
    <div
      class="property-card__email"
    >
      stub@email.com
    </div>
    <div
      class="property-card__price"
    >
      120,000
    </div>
    <div
      class="property-card__title"
    >
      Stub title
    </div>
    <div
      class="property-card__type"
    >
      flat
    </div>
  </div>
</DocumentFragment>
`;
